render/preprocessor/control/case.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package control
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/asteris-llc/converge/parse"
+	"github.com/pkg/errors"
+)
+
+// Case represents a case structure from a switch element.  Each case may have
+// multiple nodes that will be expanded from the predicate.
+type Case struct {
+	Name       string
+	Predicate  string
+	InnerNodes []*parse.Node
+}
+
+// GenerateNode generates a parse.Node for the macro-expanded placeholde from
+// the case clause
+func (c *Case) GenerateNode() (*parse.Node, error) {
+	switchHCL := fmt.Sprintf("macro.case %q {\n", c.Name)
```

Comment:
Creating an `ast.Node` from scratch was a lot more fiddly than I anticipated, and it made sense for the moment to just go ahead and `sprintf` the hcl and load it.  I think we should have a separate issue to refactor, this was just for the first pass.